### ISSUE TITLE
Environment checking

### ISSUE THREAD

**Issue by savente93** [prefix-dev/pixi#181]
I work with a lot of people that have very long living environments, so it would be quite nice if there was a way whether a current environment is compatible with a lockfile. Here I'm pulling inspiration from `poetry check` which will check if the versions currently installed are compatible/equal (I'm not actually sure which one) with those listed in the lockfile.

**Comment by savente93** [comment]
Sorry it took me so long to get back to you, that was a little rude of me. So let me try to illustrate what I mean with an example. We have a repo called X and on it work three people, A, B and C. To make sure they all have consistent envs with eachother they check in the lock file with the repository. 

- First A decides to run `pixi add tomli` which installs it to their env and write some tests for it. They add the lockfile and push to the repo.
- When B now pulls the repo they have to reinstall/ update their environment, which it would be nice to be able to check without having to reresolve everything (if that's even possible, I assume that's what poetry does) 
- meanwhile C added polars to the project for some small automation but forgot to check in the lock file, so now when the CI tries to install from the lock file it breaks. For this one specifically, at a previous job we had a CI that checked if the lockfile and the pyproject.toml were still in sync which was required before merging to main. 

Basically what I want is to have a lightweight way to check for sync errors between multiple actors. I hope this clears it up a bit, let me know if you still have questions

**Comment by baszalmstra** [comment]
Hey @savente93! No worries at all :)

I think for pixi things are a little different. For every command that interacts with an environment (`pixi run`, `pixi add`, `pixi shell`, etc.) pixi will *always* update the lock-file (if it is not in sync). This check is relatively cheap. This ensures that the lock-file is always in sync with the project file. If someone forgot to check in the lock-file, the lock-file is simply generated on the next persons' machine.

I don't think the case for person C can happen. Worst case if the lock-file is not checked in, it is regenerated in CI before running any tests.

You can see this by doing:

```shell
pixi init bla
cd bla
pixi add python
pixi run python
rm pixi.lock # Now remove the lock-file
pixi run python
# The lock file is back!
```

I think a lot of package managers do have a `--locked`/`--frozen-lockfile`/`--frozen` flag though that is used in CI. This will cause the program to error out if the lock-file is not up to date with the project file instead of regenerating the lock.

Maybe I still misunderstand the use case, or does this help?

**Comment by savente93** [comment]
Yeah I think this does help I think I should be good then. Thanks for the explanation! :)
